### DISCUSSION TITLE
Using accordion

### DISCUSSION THREAD

**Discussion by tteke** [chakra-ui/chakra-ui#1979] [General] [answered]
How can we use the onChange function on the accordion properly? If I put onChange on AccordionItem nothing happens, If I put it on accordion itself, it just returns an id (number). Giving an id prop to the AccordionItem does not alter the callback of onchange on the accordion, i.e. still returns the internal number based id.
I am asking this because I have multiple accordions stacked on a page and I want to fetch their content only when one is expanded, using onChange seemed to be the most obvious solution but I cant seem to make it work properly

**Comment by with-heart** [comment, marked as answer]
Hi @tteke! 

There is no `onChange` prop for `AccordionItem` as of the last release. That's handled internally and is provided to the items by the `Accordion`'s `Context`. We forgot to remove that from the documentation, so I opened #1983 to update the docs on that.

The `onChange` prop for `Accordion` returns either the `index` of the `AccordionItem` being opened or—if `allowMultiple` is enabled—the array of indexes that are open.

Do you mean that you want to fetch the content of each item from an API?

**Reply by tteke** [reply]
Yes thats exactly what I aim to do. And I want to do that when the AccordionItem is expanded. Also accordionItems are in a map. So if accordion's onChange returns the index, I should use the index in the array's map function right?

**Comment by chasinhues** [comment]
Resolving this discussion due to inactivity.
